src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -769,8 +772,11 @@ public void deleteResultsAndSubmissionsOfParticipation(Long participationId, boo
         resultsToBeDeleted.forEach(result -> resultService.deleteResult(result, false));
         // Delete all submissions for this participation
         submissions.forEach(submission -> {
-            if (submission instanceof ProgrammingSubmission) {
+            if (submission instanceof ProgrammingSubmission programmingSubmission) {
                 coverageReportRepository.deleteBySubmissionId(submission.getId());
+                // This clears the build log entries and deletes them from the corresponding table
+                programmingSubmission.setBuildLogEntries(Collections.emptyList());
+                submissionRepository.save(submission);
```

Comment:
Do you need to save again even if it is deleted directly afterwards in line 782?

Reply:
Yes, because otherwise I encountered issues with the OneToMany-mapping.